src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.ValidateToken.Internal.cs

```diff
@@ -63,6 +63,13 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
                 return validatedIssuerResult.UnwrapError().AddStackFrame(StackFrames.IssuerValidationFailed);
             }
 
+            var signatureValidationResult = ValidateSignature(samlToken, validationParameters, callContext);
+            if (!signatureValidationResult.IsValid)
+            {
+                StackFrames.SignatureValidationFailed ??= new StackFrame(true);
```

Comment:
When would this already be set such that we'd end up here?

Reply:
I am not sure I follow the question. If the signature validation fails, we ensure the stack frame is created, then we return the error including it.
We are moving the caching logic away in a separate PR.